src/base/show.rs

```diff
@@ -0,0 +1,284 @@
+use core::fmt;
+
+#[derive(Clone, Copy, Debug)]
+pub struct Error;
+
+impl From<fmt::Error> for Error {
+    fn from(_: fmt::Error) -> Self {
+        Self
+    }
+}
+
+pub type Result = std::result::Result<(), Error>;
+
+/// Show a value as zonefile format
+pub trait Show {
```

Comment:
Why not pick a name that has something to do with zonefile format. How would anyone guess that 'Show' means zonefile format? It's not like the name has to be short. For example, ToZonefileFormat would be a lot more desciptive.

Reply:
The trait creates data in presentation format and I think ‘show’ is a pretty good verb for that. It is also consistent with the related traits in the crate. We have “parse“ and “compose” for dealing with wire format and “scan“ and now “show” for presentation format.

Reply:
I disagree with your notion that names must be (or, indeed, can be) fully self describing. That sort of thing leads to long, complicated names and people having assumptions about how things work.

These concepts are very fundamental to what this crate is providing, so I think giving them short designators and provide a thorough definition of what they do is a good approach. You look this up in the documentation and then remember that “show” means printing in presentation format.

Reply:
Just a lot of friction for people who are not an expert in using the domain crate.  Inventing terminology that is only used in one piece of software tends to make people want to avoid that software.

The whole parse, compose, and scan are bad enough. Time to move to something more sensible.

Obviously, if a name is used a lot, then it needs to be short, even if there is some initial confusion. But show would be rare enough for most people that it will just feel awkward.

Maybe we only want to write software for people who are experts in using the domain crate. In that case using random names is fine.

Reply:
It has taken until now to discover that we don't actually have a way to write zonefile format and that using Display for that has issues.

So it seems that this feature may be fundamental, but is not commonly used. Features that are rarely used benefit from longer names that are more descriptive. The whole point of this feature is to generate zonefile format. So the name can include that without being overly specific.

This feature is introduced because the intent is to not show the output (to a user). We already have Display for showing output. This feature is a specific text-based encoding that is meant to be used by other software. Obviously, expert users can look at text-based encodings. But that is not the main purpose. Some fields that could have been mnemonic have to be output as numbers because that is what zonefile format requires. Again a good hint that the purpose is not to show that output to a user.